### PR TITLE
Error handling if broadcast events could not be sent

### PR DESCRIPTION
**Pull request by weeman1337**
Closes https://github.com/vector-im/element-web/issues/24141

Should probably better be reviewed commit-wise.

If a connection error occurs:
- Set an error state
- Pause the recorder

conn_err

If the connection is back: Broadcast should be in paused state.

PSF-1833

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: src/voice-broadcast/models/VoiceBroadcastRecording.ts [hunk 1]
```diff
@@ -248,56 +320,102 @@ export class VoiceBroadcastRecording
...
-            sequence: ++this.sequence,
+            content["io.element.voice_broadcast_chunk"] = {
+                sequence: ++this.sequence,
+            };
+
+            await this.client.sendMessage(this.roomId, content);
         };
 
-        await this.client.sendMessage(this.infoEvent.getRoomId(), content);
+        try {
```
Comment (kerryarchibald) [inline src/voice-broadcast/models/VoiceBroadcastRecording.ts:377]: Could this try catch be abstracted into a wrapping function? something like
```
private async callWithRetry(function) => {
    try {
        await function();
    catch {
       this.toRetry.push(function);
       this.onConnectionError
    }
}
// then something like:
await this.callWithRetry(() => this.sendVoiceMessage(props));

```

**Review by kerryarchibald — COMMENTED** [review]
Mostly looks good, just not sure about the chunk sequence on retry

File: src/voice-broadcast/models/VoiceBroadcastRecording.ts [hunk 2]
```diff
@@ -248,56 +320,102 @@ export class VoiceBroadcastRecording
...
+                file,
+            );
+            content["m.relates_to"] = {
+                rel_type: RelationType.Reference,
+                event_id: this.infoEventId,
+            };
             /** Increment the last sequence number and use it for this message. Also see {@link sequence}. */
-            sequence: ++this.sequence,
+            content["io.element.voice_broadcast_chunk"] = {
+                sequence: ++this.sequence,
```
[the remark this answers is not in the document]
Comment (weeman1337) [reply src/voice-broadcast/models/VoiceBroadcastRecording.ts:371]: :scream: good catch. Sequence should definitively be outside of the retry function. Tests also need updates to cover this.

File: src/voice-broadcast/models/VoiceBroadcastRecording.ts (hunk 1 above)
Comment (weeman1337) [reply src/voice-broadcast/models/VoiceBroadcastRecording.ts]: Factored out `callWithRetry` :heavy_check_mark:

File: src/voice-broadcast/models/VoiceBroadcastRecording.ts [hunk 3]
```diff
@@ -366,21 +367,14 @@ export class VoiceBroadcastRecording
...
         };
 
-        try {
-            await sendMessageFn();
-        } catch {
-            // Sending the voice message failed.
-            this.toRetry.push(sendMessageFn);
-            this.onConnectionError();
-        }
+        this.callWithRetry(sendMessageFn);
```
Comment (kerryarchibald) [inline src/voice-broadcast/models/VoiceBroadcastRecording.ts:377]: Should this be returned or awaited?

**Review by kerryarchibald — APPROVED** [review]
LGTM, left one small question

**Pull request merged**